macros/generate_audit_report_for_folder.sql

```diff
@@ -0,0 +1,31 @@
+{% macro generate_audit_report_for_folder(model_folder_path, compare_database, compare_schema, identifiers = {}, primary_keys = {}, exclude_columns = {}, audit_helper_macro_name = 'compare_relations') %}
+
+    {%- set re = modules.re -%}
+
+    {#- *----- Create a list to hold the model nodes -----* -#}
+    {%- set model_list = [] %}
+
+    {#- *-------------------------------------------------
+        * Get the list of models in the resource 
+        * path and add them to the list 
+    ------------------------------------------------------* -#}
+    {%- if execute %}
+        {%- for node in graph.nodes.values() | selectattr("resource_type", "equalto", "model") %}
+            {% set regex_folder_pattern = 'models/' ~ model_folder_path ~ '/.*' %}
```

Comment:
This works for Mac/Unix (incl dbt Cloud), but will not work for Windows which uses `\` as delimitor. We have a macro in dbt_project_evaluator to return the folder separator based on the OS.

Reply:
You're so right - do you think we should move this macro into dbt_utils?

Reply:
It feels like it could, but `dbt_utils` feels odd as well. I think that it would make even more sense to add it to `dbt-core`. I kind of recall a Slack thread where we talked about it (I can't find it again though). The current approach is ok but quite convoluted, ideally, we would just want to return [os.sep](https://docs.python.org/3/library/os.html#os.sep).

Reply:
Is os.sep not available in dbt jinja? What's blocking us from doing the simpler approach?

Reply:
`os.sep` is a Python syntax. We would need to write some code to make it available in dbt Core in Jinja, in something like [here](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/context/base.py)

Reply:
Should I open up an issue in core? What's the short term solution here - adding the project eval macro to this package as well?

Reply:
I think an issue in core could be good yes, mentioning the current use cases for the workaround.